spackbot/handlers/style.py

```diff
@@ -23,3 +30,98 @@ async def style_comment(event, gh):
                 number,
             )
             await gh.post(comments_url, {}, data={"body": comments.style_message})
+
+
+def is_up_to_date(output):
+    """
+    A commit can fail if there are no changes!
+    """
+    return "branch is up to date" in output
+
+
+async def fix_style(event, gh):
+    """
+    Respond to a request to fix style.
+    We first retrieve metadata about the pull request. If the request comes
+    from anyone with write access to the repository, we commit, and we commit
+    under the identity of the original person that opened the PR.
+    """
+    response = requests.get(event.data["issue"]["pull_request"]["url"])
+    pr = response.json()
+
+    # Get the sender of the PR - do they have write?
+    sender = event.data["sender"]["login"]
+    repository = event.data["repository"]
+    collaborators_url = repository["collaborators_url"]
+
+    # If they don't have write, we don't allow the command
+    if not await helpers.found(gh.getitem(collaborators_url, {"collaborator": sender})):
+        logger.info(f"Not found: {sender}")
+        return (
+            "Sorry %s, I cannot do that for you. Only users with write can make this request!"
+            % sender
+        )
+
+    # Tell the user the style fix is going to take a minute or two
+    message = "Let me see if I can fix that for you! This might take a moment..."
+    await gh.post(event.data["issue"]["comments_url"], {}, data={"body": message})
+
+    # Get the username of the original committer
+    user = pr["user"]["login"]
+    email = "%s@users.noreply.github.com" % user
```

Comment:
I think either will still work for the commit, at least I've had both work for me. Are you sure we need to add this extra endpoint?

Reply:
If both work for you then I think we could unconditionally use the id -- your account was created before July 18, 2017.

If we omit the id, it can end up that we do the commit on behalf of the wrong user (for users where there was some previous incarnation of themselves).

I think that we need to get the user's id somehow regardless.  Curious if you can get that from the `pr` dictionary.

Reply:
Ok this is added.